src/types.ts

```diff
@@ -12,30 +12,140 @@ export interface PlayerType {
   playerIndex: number;
   port: number;
   characterId: number | null;
-  characterColor: number | null;
-  startStocks: number | null;
   type: number | null;
+  startStocks: number | null;
+  characterColor: number | null;
+  teamShade: number | null;
+  handicap: number | null;
   teamId: number | null;
+  staminaMode: boolean | null;
+  silentCharacter: boolean | null;
+  invisible: boolean | null;
+  lowGravity: boolean | null;
+  blackStockIcon: boolean | null;
+  metal: boolean | null;
+  startOnAngelPlatform: boolean | null;
+  rumbleEnabled: boolean | null;
+  cpuLevel: number | null;
+  offenseRatio: number | null;
+  defenseRatio: number | null;
+  modelScale: number | null;
   controllerFix: string | null;
   nametag: string | null;
   displayName: string;
   connectCode: string;
   userId: string;
 }
 
-export enum GameMode {
+export enum GameModeType {
   VS = 0x02,
+  CLASSIC = 0x03,
+  ADVENTURE = 0x04,
+  ALL_STAR = 0x05,
```

Comment:
@JLaferri, I remember asking you about this before, about whether or not the contants exported in `slippi-js` should be exhaustive and represent a general Melee JS library or Slippi-specific JS library, and I recall you said something along the lines of Slippi and original Melee has divered in some ways so you'd prefer it to just be Slippi-specific constants e.g. 0x08 is not used normally, so we use it for online.

I don't know if you recall this conversation and whether or not you have thoughts about this PR expanding all these constants to include all the original Melee modes/constants.

Reply:
As part of my exploration of stadium replays, I ended up recording replays for every game mode. A lot of them don't work properly, but there are plenty in other modes that already work; I figured I'd add the rest while I was here since it's possible I work on other modes down the line - it's fine to add them when we officially support those replays though.